### PR TITLE
Makefile: natively build without setting platform

### LINKED ISSUE #2450: nightly build failed
As the build server from PingCAP doesn't start docker daemon with experiment feature, it cannot build chaos-mesh with `--platform`. The nightly build has failed for a month.

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

close #2450 

### What's changed and how it works?

Remove `--platform` arguments for native build. I really hope it doesn't break the arm developer environment. @iguoyr PTAL.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -131,23 +131,26 @@ endif
 GO_TARGET_PHONY += $(1)
 endef
 
-ifeq ($(TARGET_PLATFORM),)
+BUILD_INDOCKER_ARG := --env IN_DOCKER=1 --env HTTP_PROXY=${HTTP_PROXY} --env HTTPS_PROXY=${HTTPS_PROXY} --env GOPROXY=${GOPROXY} --volume $(ROOT):/mnt --user $(shell id -u):$(shell id -g)
+
+ifneq ($(TARGET_PLATFORM),)
+	BUILD_INDOCKER_ARG += --platform=linux/$(TARGET_PLATFORM)
+	DOCKER_BUILD_ARGS += --platform=linux/$(TARGET_PLATFORM) --build-arg TARGET_PLATFORM=amd64
```
Comment (iguoyr) [inline Makefile:138]: `--build-arg TARGET_PLATFORM=amd64`, why not use `$TARGET_PLATFORM` here?

Comment (YangKeao) [reply Makefile]: Sorry. It's a typo :(

**Review by iguoyr — APPROVED** [review]
LGTM

**Review by STRRL — APPROVED** [review]
LGTM!

**Pull request merged**